src/core/packages/root/browser-internal/src/filters/ebt_span_filter.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+const ignorePaths = ['/kibana-browser'];
+
+// Copied from @elastic/apm-rum/src/common/types.ts
+export type FilterFn = (payload: Payload) => Payload | boolean | void;
+export interface Payload {
+  [key: string]: any;
+}
+
+const hasTrackedSpan = (spans: any): boolean => {
+  // Check if there is any span with a URL other than ignorePaths
+  return !spans.every((span: any) => {
+    const url = span?.context?.http?.url;
+    return typeof url === 'string' ? ignorePaths.some((p) => url.includes(p)) : false;
+  });
+};
+
+// Related to https://github.com/elastic/observability-dev/issues/4529
+// This filter is only applied to transactions of type 'user-interaction'.
+// If a 'user-interaction' transaction contains only EBT spans, it will be filtered out.
+export const ebtSpanFilter: FilterFn = (payload: Payload) => {
```

Comment:
I started implementing this logic by only filtering EBT spans, but since RUM discards 'user-interaction' transactions with no spans before this filter, I changed this filter to also discard transactions.

Now, I am thinking about changing the name to `discardEBTOnlyTx'. If you have a different suggestion, please let me know.

Reply:
We'll still report the EBT span in transactions that include the EBT span but have more spans? I wonder if that can cause bad measures:

```
Transaction 1 (full duration > 1s instead of 200ms)
  One sub-action (100ms)
    Another sub-action (100ms)
                        (EBT 1s delay)         EBT span
```

Maybe we should always discard the span, but leave the rest of the spans in the transaction. Then discard empty transactions.

WDYT?

Reply:
Interesting! Thanks for confirming!